File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        for(int i=1;i<arr.length;i++) {
            answer = gcd(arr[i-1], arr[i]);
            arr[i] = (arr[i-1] * arr[i]) / answer;
            System.out.println(arr[i]);
        }
        answer = arr[arr.length-1];
        return answer;
    }
    public int gcd(int a, int b) {
        while(b !=0) {
            int tmp = b;
            b = a % b;
            a = tmp;
        }
        return a;
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    var answer = 0;
    function gcd(a,b) { //최대공약수
        while(b!==0) {
            const tmp = b;
            b = a % b 
            a = tmp
        }
        return a
    }
    function lcm(a,b) {
       return a*b / gcd(a,b)
    }
    answer = arr.reduce((a,b)=> lcm(a,b))
    return answer;
}